GVFS/GVFS.Common/Paths.Shared.cs

```diff
@@ -13,10 +13,24 @@ public static string GetGitEnlistmentRoot(string directory)
 
         public static string GetServiceDataRoot(string serviceName)
         {
-            return Path.Combine(
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+            {
+                return Path.Combine(
                 Environment.GetFolderPath(Environment.SpecialFolder.CommonApplicationData, Environment.SpecialFolderOption.Create),
                 "GVFS",
                 serviceName);
+            }
+
+            string path = Path.Combine(
+                Environment.GetEnvironmentVariable("HOME"),
+                "Library",
+                "Application Support",
+                "GVFS",
+                serviceName);
+
+            Directory.CreateDirectory(path);
```

Comment:
Does this method `GetServiceDataRoot` need to create a directory? I think this might be a surprising side effect of a `Getter` method...

Reply:
My bad... Removed it and moved the `GetServiceDataRoot` to `GVFSPlatform`.